capstone-frontend/src/components/Home.jsx

```diff
@@ -0,0 +1,304 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom'
+import { CardDeck, Tabs, Tab, Card, Spinner, Button } from 'react-bootstrap';
+import ClubGridItem from './ClubGridItem.jsx'
+import { BookDescriptionOverlay } from "./BookDescriptionOverlay";
+import CreateList from './CreateList'
+import '../App.css';
+import '../styles/Home.css'
+
+class Home extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      clubsOwned: [],
+      clubsMember: [],
+      myBookListsMap: new Map(),
+      sharedBookListsMap: new Map(),
+      key: 'Clubs - Owner', // For Bootstrap Tabs
+      screenWidth: window.outerWidth,
+      fetchingClubsOwned: true, // For Spinner
+      fetchingClubsMember: true, // For Spinner
+      fetchingBookListsOwned: true, // For Spinner
+      fetchingBookListsShared: true, // For Spinner
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("resize", this.resize);
+    this.fetchClubs('own');
+    this.fetchClubs('shared');
+    this.fetchBookLists('own');
+    this.fetchBookLists('shared');
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.resize);
+  }
+
+  resize = () => {
+    this.setState({ screenWidth: window.outerWidth });
+
+    this.fetchBookLists('own');
+    this.fetchBookLists('shared');
+  }
+
+  fetchClubs = async (type) => {
+
+    const userID = window.localStorage.getItem('userID');
+
+    if (type === 'own') {
+      this.setState({ fetchingClubsOwned: true });
+      const clubsOwned = await fetch(`/api/clubs?ownerID=${userID}`).then(resp => resp.json());
```

Comment:
Should this have a `.catch` appended to it in the case of errors?

Reply:
Ok